staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -2874,6 +2874,229 @@ metadata:
 `, string(updatedCRYaml))
 }
 
+func TestApplyWithPruneV2ApplySetNamespace(t *testing.T) {
+	tf := cmdtesting.NewTestFactory()
+	defer tf.Cleanup()
+
+	scheme := runtime.NewScheme()
+
+	listMapping := map[schema.GroupVersionResource]string{
+		{Group: "", Version: "v1", Resource: "namespaces"}: "NamespaceList",
+	}
+
+	fakeDynamicClient := dynamicfakeclient.NewSimpleDynamicClientWithCustomListKinds(scheme, listMapping)
+	tf.FakeDynamicClient = fakeDynamicClient
+	tf.UnstructuredClient = &fake.RESTClient{
+		NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
```

Comment:
Maybe it's time to refactor our test clients - though I suggest not in this PR.

This assumes that we can't merge with TestApplyWithPruneV2; though I would say the fact that I can't easily tell whether they are the same is itself a good argument for refactoring a common test client!

cc @soltysh as I think you were not convinced in the past.